charcoal/utils.py

```diff
@@ -153,7 +153,12 @@ def gather_at_rank(mh, lca_db, lin_db, match_rank):
     # do the gather:
     counts = Counter()
     while 1:
+        
+        if len(query_sig.minhash.hashes) == 0:
```

Comment:
can use `if not query_sig.minhash:` here. note also `len(query_sig.minhash)` is the same as `len(query_sig.minhash.hashes)`.